File: backend/app/schemas/documents.py
from datetime import datetime
from typing import Any
from uuid import UUID

from pydantic import BaseModel, Field


class DocumentUploadResponse(BaseModel):
    document_id: UUID
    file_name: str
    status: str


class DocumentListItem(BaseModel):
    id: UUID
    file_name: str
    file_type: str
    status: str
    chunk_count: int
    created_at: datetime
    error_message: str | None = None


class DocumentListResponse(BaseModel):
    documents: list[DocumentListItem]


class DocumentDetailResponse(BaseModel):
    id: UUID
    file_name: str
    file_type: str
    status: str
    chunk_count: int
    created_at: datetime
    updated_at: datetime
    error_message: str | None = None
    chunks: list[dict[str, Any]] = Field(default_factory=list)


class DocumentDeleteResponse(BaseModel):
    document_id: UUID
    deleted: bool
    deleted_agent_runs: int
    deleted_agent_steps: int
    deleted_chat_messages: int
    deleted_chat_sessions: int
    deleted_chunks: int
    deleted_entities: int
    deleted_relationships: int
    deleted_qdrant_points: bool
    deleted_storage_file: bool

File: frontend/src/types/documents.ts
export type DocumentStatus =
  | "uploaded"
  | "processing"
  | "ready"
  | "failed"
  | "deleting";

export type JsonValue =
  | string
  | number
  | boolean
  | null
  | JsonValue[]
  | { [key: string]: JsonValue };

export type DocumentChunk = {
  [key: string]: JsonValue;
};

export type DocumentListItem = {
  id: string;
  file_name: string;
  file_type: string;
  status: DocumentStatus;
  chunk_count: number;
  created_at: string;
  error_message?: string | null;
};

export type DocumentListResponse = {
  documents: DocumentListItem[];
};

export type DocumentUploadResponse = {
  document_id: string;
  file_name: string;
  status: DocumentStatus;
};

export type DocumentDetailResponse = DocumentListItem & {
  updated_at: string;
  chunks: DocumentChunk[];
};

export type DocumentDeleteResponse = {
  document_id: string;
  deleted: true;
  deleted_agent_runs: number;
  deleted_agent_steps: number;
  deleted_chat_messages: number;
  deleted_chat_sessions: number;
  deleted_chunks: number;
  deleted_entities: number;
  deleted_relationships: number;
  deleted_qdrant_points: boolean;
  deleted_storage_file: boolean;
};
